// @4us-dev/utils/lib/utils/RandomUtils/nextString.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.specialAmbiguousCaracteres = exports.specialSimpleCaracteres = exports.numbersCaracteres = exports.lowerCaracteres = exports.upperCaracteres = void 0;
var nextStringCustom_1 = __importDefault(require("./nextStringCustom"));
exports.upperCaracteres = 'ABCDEFGHIKLMNOPQRSTVXYZ';
exports.lowerCaracteres = 'abcdefghiklmnopqrstvxyz';
exports.numbersCaracteres = '0123456789';
exports.specialSimpleCaracteres = '!#$%&*_+=-^~?;:.,|';
exports.specialAmbiguousCaracteres = '"()\'`´{[]}/><';
exports.default = (function (config) {
    var _a, _b, _c, _d, _e, _f;
    if (config === void 0) { config = {
        length: 16,
        upper: true,
        lower: true,
        numbers: true,
        specialSimple: true,
        specialAmbiguous: true,
    }; }
    var length = (_a = config.length) !== null && _a !== void 0 ? _a : 16;
    var upper = (_b = config.upper) !== null && _b !== void 0 ? _b : true;
    var lower = (_c = config.lower) !== null && _c !== void 0 ? _c : true;
    var numbers = (_d = config.numbers) !== null && _d !== void 0 ? _d : true;
    var specialSimple = (_e = config.specialSimple) !== null && _e !== void 0 ? _e : true;
    var specialAmbiguous = (_f = config.specialAmbiguous) !== null && _f !== void 0 ? _f : true;
    var caractersAllowed = '';
    if (upper)
        caractersAllowed += exports.upperCaracteres;
    if (lower)
        caractersAllowed += exports.lowerCaracteres;
    if (numbers)
        caractersAllowed += exports.numbersCaracteres;
    if (specialSimple)
        caractersAllowed += exports.specialSimpleCaracteres;
    if (specialAmbiguous)
        caractersAllowed += exports.specialAmbiguousCaracteres;
    return nextStringCustom_1.default(length, caractersAllowed);
});
