// @4us-dev/utils/lib/utils/BrazilianUtils/generateCNPJ.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BASE_CHECK_DIGIT_CNPJ = void 0;
var nextString_1 = __importDefault(require("../RandomUtils/nextString"));
var calculateCheckDigitModule11_1 = __importDefault(require("./calculateCheckDigitModule11"));
var CNPJ_SIZE_WITHOUT_FILIAL_AND_CHECK_DIGIT = 8;
var CNPJ_FILIAL = '0001';
exports.BASE_CHECK_DIGIT_CNPJ = [9, 8, 7, 6, 5, 4, 3, 2, 9, 8, 7, 6, 5];
exports.default = (function () {
    var cnpjWithoutCheckDigit = nextString_1.default({
        length: CNPJ_SIZE_WITHOUT_FILIAL_AND_CHECK_DIGIT,
        lower: false,
        upper: false,
        numbers: true,
        specialAmbiguous: false,
        specialSimple: false,
    });
    cnpjWithoutCheckDigit = cnpjWithoutCheckDigit + CNPJ_FILIAL;
    var firstCheckDigit = calculateCheckDigitModule11_1.default(cnpjWithoutCheckDigit, exports.BASE_CHECK_DIGIT_CNPJ);
    var secondCheckDigit = calculateCheckDigitModule11_1.default("" + cnpjWithoutCheckDigit + firstCheckDigit, exports.BASE_CHECK_DIGIT_CNPJ);
    return "" + cnpjWithoutCheckDigit + firstCheckDigit + secondCheckDigit;
});
