// @4us-dev/utils/lib/utils/BrazilianUtils/generateCPF.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BASE_CHECK_DIGIT_CPF = void 0;
var nextString_1 = __importDefault(require("../RandomUtils/nextString"));
var calculateCheckDigitModule11_1 = __importDefault(require("./calculateCheckDigitModule11"));
var CPF_SIZE_WITHOUT_CHECK_DIGIT = 9;
exports.BASE_CHECK_DIGIT_CPF = [9, 8, 7, 6, 5, 4, 3, 2, 1, 0];
exports.default = (function () {
    var cpfWithoutCheckDigit = nextString_1.default({
        length: CPF_SIZE_WITHOUT_CHECK_DIGIT,
        lower: false,
        upper: false,
        numbers: true,
        specialAmbiguous: false,
        specialSimple: false,
    });
    var firstCheckDigit = calculateCheckDigitModule11_1.default(cpfWithoutCheckDigit, exports.BASE_CHECK_DIGIT_CPF);
    var secondCheckDigit = calculateCheckDigitModule11_1.default("" + cpfWithoutCheckDigit + firstCheckDigit, exports.BASE_CHECK_DIGIT_CPF);
    return "" + cpfWithoutCheckDigit + firstCheckDigit + secondCheckDigit;
});
